crates/extensions/tedge_health_ext/src/lib.rs

```diff
@@ -37,12 +44,65 @@ impl HealthMonitorBuilder {
         .try_into()
         .expect("Failed to create the HealthMonitorActor topic filter");
 
+        let service_health_topic =
+            ServiceHealthTopic::from_old_topic(format!("tedge/health/{service_name}")).unwrap();
+
         let mut box_builder = SimpleMessageBoxBuilder::new(service_name, 16);
         box_builder
             .set_request_sender(mqtt.connect_consumer(subscriptions, box_builder.get_sender()));
 
         let builder = HealthMonitorBuilder {
-            service_name: service_name.to_owned(),
+            service_health_topic,
+            box_builder,
+        };
+
+        // Update the MQTT config
+        *mqtt.as_mut() = builder.set_init_and_last_will(mqtt.as_mut().clone());
+
+        builder
+    }
+
+    /// Creates a HealthMonitorBuilder that creates a HealthMonitorActor with
+    /// a new topic scheme.
+    pub fn from_service_topic_id(
+        service: ServiceTopicId,
```

Comment:
It would be far better to pass the`device` topic id along the `service` topic id, instead of  extracting the former from the latter. Indeed, there is no guaranties that the topic scheme will be the default. 

```suggestion
    /// Creates a HealthMonitorBuilder that creates a HealthMonitorActor with
    /// a new topic scheme.
    pub fn from_service_topic_id(
        device: EntityTopicId,
        service: ServiceTopicId,
```

Reply:
You're right, it makes no sense to discard device topic id just to derive it later, but instead of adding another argument to the function, it would be better to embed a device topic id inside a `ServiceTopicId` type, so every service contains information about its associated device.

Reply:
Yes, `ServiceTopicId` starts to make sense when embedding the device topic id.